src/components/legend/draw.js

```diff
@@ -184,27 +184,42 @@ module.exports = function draw(gd) {
     if(anchorUtils.isRightAnchor(opts)) {
         lx -= opts.width;
     }
-    if(anchorUtils.isCenterAnchor(opts)) {
+    else if(anchorUtils.isCenterAnchor(opts)) {
         lx -= opts.width / 2;
     }
 
     if(anchorUtils.isBottomAnchor(opts)) {
         ly -= opts.height;
     }
-    if(anchorUtils.isMiddleAnchor(opts)) {
+    else if(anchorUtils.isMiddleAnchor(opts)) {
         ly -= opts.height / 2;
     }
 
+    lx = Math.round(lx);
+    ly = Math.round(ly);
+
+    // Make sure the legend top is below the top margin
+    if(ly < fullLayout.margin.t) ly = fullLayout.margin.t;
+
+    var scrollHeightMax = fullLayout.height - fullLayout.margin.b - ly;
+    var scrollHeight = Math.min(scrollHeightMax, opts.height);
+
+    if(scrollHeight <= 2 * opts.borderwidth) {
+        console.error('Legend.draw: insufficient space to draw legend');
+        legend.remove();
+        return;
+    }
```

Comment:
On second thoughts, it'd be more user-friendly to provide a safe default here. For example:
- if `opts.height >= fullLayout.height`, then `ly = fullLayout.margin.t; scrollHeight = fullLayout.height;`
- else locate the legend the bottom.